### PR TITLE
Adding plot_carpet function

### LINKED ISSUE #1585: Add function for creating "carpet plots"
image

Potentially ported from MRIQC (see https://github.com/poldracklab/mriqc/issues/674)

### PR DESCRIPTION
**Pull request by chrisgorgo**
Closes #1585.
image

Most of this code has been written by @oesteban for MRIQC.
This is the initial most basic implementation. More features can be added in subsequent PRs.

What is the expected amount of tests/documentation/examples to get this merged?

### CODE REVIEW

**Comment by arielletambini** [comment]
I've tried to use the plot_carpet function without providing a mask image (which is optional) - but get the following error (mask_img_ is not found from NiftiMasker):

In [12]: plotting.plot_carpet(img)
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-12-bd2c8c24e6be> in <module>()
----> 1 plotting.plot_carpet(img)

/Applications/anaconda3/lib/python3.6/site-packages/nilearn/plotting/img_plotting.py in plot_carpet(img, mask_img, detrend, output_file, figure, axes, title)
   1308     if not mask_img:
   1309         nifti_masker = NiftiMasker(mask_strategy='epi', standardize=False)
-> 1310         mask_data = nifti_masker.mask_img_.get_data().astype(bool)
   1311     else:
   1312         mask_nii = _utils.check_niimg_3d(img, dtype='auto')

AttributeError: 'NiftiMasker' object has no attribute 'mask_img_'

File: nilearn/plotting/img_plotting.py [hunk 1]
```diff
@@ -1264,3 +1266,114 @@ def plot_connectome(adjacency_matrix, node_coords,
...
+            are .png, .pdf, .svg. If output_file is not None, the plot
+            is saved to a file, and the display is closed.
+        figure : matplotlib figure, optional
+            Matplotlib figure used. If None is given, a
+            new figure is created.
+        axes : matplotlib axes, optional
+            The axes used to display the plot. If None, the complete
+            figure is used.
+        title : string, optional
+            The title displayed on the figure.
```
Comment (KamalakerDadi) [inline nilearn/plotting/img_plotting.py]: Missing Returns

**Comment by KamalakerDadi** [comment]
Could you please add tests ? As we have 0.3% decrease in coverage.

**Comment by KamalakerDadi** [comment]
Adding an example should be great too. Basically, demonstrating how to use this new cool feature.

**Comment by GaelVaroquaux** [comment]
> Adding an example should be great too. Basically, demonstrating how to
> use this new cool feature.

Yes, an example is a necessary to add any functionality to nilearn,
because it is a teaching vehicle.

But here it should be easy: the first subject of
datasets.fetch_adhd(n_subjects=1) has a movement confound that is very
visible on the time-series, so it should make a good example.

**Comment by tsalo** [comment]
Since #2456 was build off of this and was merged, can this be closed?

**Comment by emdupre** [comment]
I think so ! Thanks for following up, and thanks @chrisgorgo for your work on this !

**Pull request closed without merging**